workspaces/mcp-chat-client/plugins/mcp-chat-client/src/components/RightPane/RightPane.tsx

```diff
@@ -0,0 +1,329 @@
+/*
+ * Copyright 2025 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, { useEffect, useState } from 'react';
+import { useTheme } from '@mui/material/styles';
+import AddIcon from '@mui/icons-material/Add';
+import ChevronLeftIcon from '@mui/icons-material/ChevronLeft';
+import ChevronRightIcon from '@mui/icons-material/ChevronRight';
+import MemoryIcon from '@mui/icons-material/Memory';
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import IconButton from '@mui/material/IconButton';
+import Typography from '@mui/material/Typography';
+import { useApi } from '@backstage/core-plugin-api';
+import { mcpChatApiRef } from '../../api';
+import type { ConfigStatus, Tool } from '../../api/McpChatApi';
+import { ActiveMcpServers } from './ActiveMcpServers';
+import { ActiveTools } from './ActiveTools';
+import { ProviderStatus } from './ProviderStatus';
+import { BotIcon } from '../BotIcon';
+
+interface MCPServer {
+  id?: string;
+  name: string;
+  enabled: boolean;
+  type?: string;
+  hasUrl?: boolean;
+  hasNpxCommand?: boolean;
+  hasScriptPath?: boolean;
+}
+
+interface RightPaneProps {
+  sidebarCollapsed: boolean;
+  onToggleSidebar: () => void;
+  onNewChat: () => void;
+  mcpServers: MCPServer[];
+  onServerToggle: (serverName: string) => void;
+  configStatus: ConfigStatus | null;
+}
+
+export const RightPane: React.FC<RightPaneProps> = ({
+  sidebarCollapsed,
+  onToggleSidebar,
+  onNewChat,
+  mcpServers,
+  onServerToggle,
+  configStatus,
+}) => {
+  const theme = useTheme();
+  const mcpChatApi = useApi(mcpChatApiRef);
+  const [availableTools, setAvailableTools] = useState<Tool[]>([]);
+  const [toolsLoading, setToolsLoading] = useState(false);
+  const [providerConnectionStatus, setProviderConnectionStatus] = useState<{
+    connected: boolean;
+    models?: string[];
+    error?: string;
+    loading: boolean;
+  }>({
+    connected: false,
+    models: undefined,
+    error: undefined,
+    loading: true,
+  });
+
+  // Fetch available tools when component mounts or servers change
+  useEffect(() => {
```

Comment:
use `useAsync` when is possible for all the async state changes: The code gets more compact and we are safer if the component unmounts while async operations are in progress

Reply:
Thanks, it was something new for me. Explored the "react-use" library, its a great one!